### PR TITLE
Update version to v1.14.19

### PR DESCRIPTION
**Pull request by tkelman**
This isn't the latest version, but it's only about a week old.
The newer versions pin to a newer botocore than the 1.8.23 that's available in conda-forge at the moment (I'm trying to avoid needing a third PR in this series).
1.14.19 is the awscli release that corresponds to botocore 1.8.23:
https://github.com/aws/aws-cli/blob/1.14.19/setup.py#L26-L31

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ requirements:
 
   run:
     - python
-    - botocore ==1.5.83
+    - botocore ==1.8.23
     - colorama >=0.2.5,<=0.3.7
     - docutils >=0.10
     - rsa >=3.1.2,<=3.5.0
```
Comment (tkelman) [inline recipe/meta.yaml]: TODO: fix the next line to match https://github.com/aws/aws-cli/blob/1.14.19/setup.py#L30 once https://github.com/conda-forge/s3transfer-feedstock/pull/11 is merged

**Comment by tkelman** [comment]
@conda-forge-admin, please rerender

**Comment by tkelman** [comment]
ping @pmlandwehr @hajapy this should be good to merge. I'm adding myself to maintainers, hope that's fine with you. I can open a followup to update again to a version that matches the latest conda-forge version of botocore, but may as well publish this in the interim.

**Comment by ocefpaf** [comment]
Thanks @tkelman!

**Comment by isuruf** [comment]
Welcome to conda-forge @tkelman. Make sure to use PRs from forks as branches in the feedstocks are deployed.

**Comment by tkelman** [comment]
Ah good to know. I might set up some cron jobs so my forks stay up-to-date then.

**Pull request merged**